### PR TITLE
test - start code testing for login and sign up

### PR DESCRIPTION
**Pull request by bellaabdelouahab**
I also advise you to look into some importing changes.

### CODE REVIEW

File: backend-app/middlewares/globalErrorHandler.js [hunk 1]
```diff
@@ -11,13 +13,22 @@ const currentEnv = process.env.NODE_ENV || "development";
...
     Object.keys(err.keyValue).forEach((key) => {
       err.message = `Duplicate value for the field : [${key}] Please use another value!`;
     });
     err.statusCode = 409;
   }
+  // console.log(err);
+  if (err.name === 'ValidationError') {
+      err = new AppError(400, 'fail', err.message);
+  }
+  // If the error dosent have a status code, set it to 500
```
Comment (muttaqin1) [inline backend-app/middlewares/globalErrorHandler.js:27]: We can check if the error is a  instance of our AppError class. Then we can easily filter our api specific errors and handle them seperately .And the other errors will be handled with 500 - internal server error

**Review by Khalid1G — COMMENTED** [review]
good job, 
you can write test for the function using to verify JWT token
proposed cases: 

- normal case.
- no token provide
- invalid token (malformed) 
- change password and use the same token that generated before change password
- delete use and use there token
...

**Review by muttaqin1 — APPROVED** [review]

File: backend-app/middlewares/globalErrorHandler.js (hunk 1 above)
Comment (Khalid1G) [reply backend-app/middlewares/globalErrorHandler.js:27]: actually we do that in the global middleware error.

```js
const sendErrorProd = (err, res) => {
  if (err.isOperational) {
    res.status(err.statusCode).json({
      status: err.status,
      message: err.message,
      errors: err.errors,
    });
  } else {
    res.status(500).json({
      status: "error",
      message: "Something went very wrong!",
    });
  }
};
```

@muttaqin1

Comment (muttaqin1) [reply backend-app/middlewares/globalErrorHandler.js:27]: I think this is the best way to handle errors. I filter api specific errors using this function in my project:
```ts
export const IsApiError = (err: unknown): err is ApiError =>
  err instanceof ApiError ? err.isOperationalError : false;
```
@Khalid1G

Comment (bellaabdelouahab) [reply backend-app/middlewares/globalErrorHandler.js:27]: can you make a pr with those changes @muttaqin1

Comment (muttaqin1) [reply backend-app/middlewares/globalErrorHandler.js:27]: yeah sure.
@bellaabdelouahab

**Pull request merged**